torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -149,8 +149,10 @@ at::Tensor tensor_from_numpy(PyObject* obj) {
   for (int i = 0; i < ndim; i++) {
     if (strides[i] < 0) {
       throw ValueError(
-          "some of the strides of a given numpy array are negative. This is "
-          "currently not supported, but will be added in future releases.");
+          "At least one stride in the given numpy array is negative, and "
+          "tensors with negative strides are not currently supported. "
+          "You can make your numpy array contiguous with np.ascontiguousarray() "
+          "to workaround this issue. (Doing so may copy your data.) ");
```

Comment:
`may`? If the stride is negative, it will always copy no?

Reply:
Sure. We can be more definitive ;)

Reply:
Not only does it not always copy, it also does not always work 
```
In [34]: a=np.arange(12).reshape(1,12)[::-1]                                                                                           

In [35]: b=np.ascontiguousarray(a) #no copy                                                                                            

In [36]: at = torch.from_numpy(np.ascontiguousarray(a)) # does not work                                                                
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-36-f588971b084b> in <module>
----> 1 at = torch.from_numpy(np.ascontiguousarray(a)) # does not work

ValueError: some of the strides of a given numpy array are negative. This is currently not supported, but will be added in future releases.
```

Reply:
:o 

Would `b = np.copy(a, order='C')` do the job then?